v0/destinations/zendesk/transform.js

```diff
@@ -350,15 +453,41 @@ async function processIdentify(message, destinationConfig, headers) {
   const url = endPoint + category.createOrUpdateUserEndpoint;
   const returnList = [];
 
-  const traits = getFieldValueFromMessage(message, "traits");
+  const userId = await getUserId(message, headers);
+  // If primaryEmail and userId both exists then it will set it as primary email.
+  if (primaryEmail && userId) {
+    const userIdentityId = await getUserIdentityId(
+      userId,
+      primaryEmail,
+      headers,
+      destinationConfig
+    );
+    if (userIdentityId) {
+      returnList.push(
+        responseBuilderToSetPrimaryAccount(userIdentityId, userId, headers)
+      );
+    }
+  } else if (primaryEmail) {
+    // if only primaryEmail exists then in the payload it will pass as user.identities
+    // and it will set it as primary while creating the user.
+    const primaryEmailPayload = {
+      type: "email",
+      value: primaryEmail
+    };
+    if (payload.user.identities && Array.isArray(payload.user.identities)) {
+      payload.user.identities.push(primaryEmailPayload);
+    } else {
+      payload.user.identities = [primaryEmailPayload];
+    }
```

Comment:
This scenario will never occur as userId is a required field if you see line number 430.